### PR TITLE
clippy: fix legacy_numeric_constants

### PR DESCRIPTION
**Pull request by yihau**
#### Problem

some changes for bumping Rust version: https://github.com/anza-xyz/agave/pull/1309

https://rust-lang.github.io/rust-clippy/master/index.html#/legacy_numeric_constants

#### Summary of Changes

```diff
- std::NUMBER_TYPE::CONSTANT_NAME
+ NUMBER_TYPE::CONSTANT_NAME
```

### CODE REVIEW

File: programs/sbf/rust/custom_heap/src/lib.rs [hunk 1]
```diff
@@ -13,7 +13,6 @@ use {
         alloc::{alloc, Layout},
         mem::{align_of, size_of},
         ptr::null_mut,
-        usize,
```
Comment (steviez) [inline programs/sbf/rust/custom_heap/src/lib.rs:16]: Is this related / strictly necessary ? A little different then the rest of the changes in this PR

**Review by steviez — COMMENTED** [review]
Just the one question

File: programs/sbf/rust/custom_heap/src/lib.rs (hunk 1 above)
Comment (yihau) [reply programs/sbf/rust/custom_heap/src/lib.rs:16]: yeah, it reported this one:
Screenshot 2024-05-14 at 14 49 47

and I don't find any constant that is used in this file so I just removed it 🙈

**Review by steviez — APPROVED** [review]
